src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,122 @@
+     */
+    public static Command parse(String command) throws DukeException {
+        Scanner sc = new Scanner(command);
+
+        if (sc.hasNext("mark")) {
+            String mark = sc.next();
+            if (!sc.hasNextInt()) {
+                throw new DukeException(NO_INDEX);
+            } else {
+                int i = sc.nextInt();
+                sc.close();
+                return new MarkCommand(i);
+            }
+        } else if (sc.hasNext("unmark")) {
+            String unmark = sc.next();
+            if (!sc.hasNextInt()) {
+                throw new DukeException(NO_INDEX);
+            } else {
+                int i = sc.nextInt();
+                sc.close();
+                return new UnmarkCommand(i);
+            }
+        } else if (sc.hasNext("deadline")) {
+            sc.useDelimiter("deadline\\s*|\\s*/by\\s*");
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_DESC_DEADLINE);
+            }
+            String description = sc.next();
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_DATE_DEADLINE);
+            }
+            String by = sc.next();
+            sc.close();
+            return new AddCommand("deadline", description, by);
+        } else if (sc.hasNext("event")) {
+            sc.useDelimiter("event\\s*|\\s*/at\\s*");
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_DESC_EVENT);
+            }
+            String description = sc.next();
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_DATE_EVENT);
+            }
+            String at = sc.next();
+            sc.close();
+            return new AddCommand("event", description, at);
+        } else if (sc.hasNext("todo")) {
+            sc.useDelimiter("todo\\s*");
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_DESC_TODO);
+            }
+            String description = sc.next();
+            sc.close();
+            return new AddCommand("todo", description, null);
+        } else if (sc.hasNext("delete")) {
+            sc.useDelimiter("delete\\s*");
+            if (!sc.hasNextInt()) {
+                throw new DukeException(NO_INDEX);
+            } else {
+                int i = sc.nextInt();
+                sc.close();
+                return new DeleteCommand(i);
+            }
+        } else if (sc.hasNext("find")) {
+            sc.useDelimiter("find\\s*");
+            if (!sc.hasNext()) {
+                throw new DukeException(NO_TARGET);
+            } else {
+                String s = sc.next();
+                sc.close();
+                return new FindCommand(s);
+            }
+        } else {
+            if (command.equals("bye")) {
+                return new ExitCommand();
+            } else if (command.equals("list")) {
+                return new ShowListCommand();
+            } else {
+                throw new DukeException(UNKNOWN_COMMAND);
```

Comment:
Nested if-else statements could be difficult to read especially when it is long. Would switch statements be a better alternative here?

Reply:
Thanks for the suggestion! I have changed it to switch and it looks neater now.